### ISSUE TITLE
Open source replays code

### ISSUE THREAD

**Issue by luisgurmendezMLabs** [haxball/haxball-issues#1107]
@basro Is there some reason why the replay code isn't open source?

It will be great to know how to interpret a hbr2 file. I have seen some github repos that make a reverse eng of the minified js of the replays page and manage to get the info that they want. Unfortunately they lack of documentation and are pretty hard to follow. I wish you can prove this to the community I know it will be very appreciated :).

**Comment by Wazarr94** [comment]
Never gonna be done in the foreseable future, but if you have time to spare, you can download the game-min.js file and reverse engineer it.

I tried it, and it's not an easy task at all, could take many days to completely decode the system.

**Comment by 9999OldSchool** [comment]
> basro decided to not make the code open source, as competitors might steal the code, I assume. Many websites tried to copy haxball, without success and I think he wants them to not be given a chance.

It would be a great idea, to play on old Haxball. Damn, we are so unlucky that we can't copy Flash Haxball and modify it a bit.

**Comment by luisgurmendezMLabs** [comment]
@Wazarr94 Yeah I understand that and I totally agree. If the code is released it will be easy to steal the game, but my request was specifically for the replay's code. I might be wrong but I don't think that the replay has any game logic in it. If that's the case it will be safe to share around as it can give enormous additional value

**Comment by davidskuza** [comment]
It has games logic in it. What would you like to extract from replay files?

**Comment by luisgurmendezMLabs** [comment]
Oh, I didn't thought it had game logic it, I'll close this issue then. 
I wanted the replays because they gives you everything about a game, you can even save the replay files and analyze them later on. This would be a pain saving this ourselves.
